include/swift/AST/Types.h

```diff
@@ -3121,8 +3121,8 @@ class AnyFunctionType : public TypeBase {
   /// Pass `selfUncurried = true` when the function type is for a method whose
   /// self parameter has been uncurried as in (A, B, C, Self) -> R.
   AnyFunctionType *getAutoDiffAssociatedFunctionType(
-      const AutoDiffParameterIndices &indices, unsigned differentiationOrder,
-      AutoDiffAssociatedFunctionKind kind,
+      const AutoDiffParameterIndices &indices, unsigned resultIndex,
```

Comment:
I will defer this change until later so that I can get this PR merged sooner. I do not think it will be much harder to make this change when we need it than to make this change now.

Reply:
Ok. That's fine.